src/isolate/backends/conda.py

```diff
@@ -34,109 +34,89 @@
 class CondaEnvironment(BaseEnvironment[Path]):
     BACKEND_NAME: ClassVar[str] = "conda"
 
-    packages: List[str] = field(default_factory=list)
+    environment_definition: Dict[str, Any] = field(default_factory=dict)
     python_version: Optional[str] = None
-    env_dict: Optional[Dict[str, Any]] = None
 
     @classmethod
     def from_config(
         cls,
         config: Dict[str, Any],
         settings: IsolateSettings = DEFAULT_SETTINGS,
     ) -> BaseEnvironment:
-        if config.get("env_dict") and config.get("env_yml_str"):
-            raise EnvironmentCreationError(
-                "Either env_dict or env_yml_str can be provided, not both!"
+        processing_config = copy.deepcopy(config)
+        processing_config.setdefault("python_version", active_python())
+
+        if "env_dict" in processing_config:
+            definition = processing_config.pop("env_dict")
+        elif "env_yml_str" in processing_config:
+            import yaml
+
+            definition = yaml.safe_load(processing_config.pop("env_yml_str"))
+        elif "packages" in processing_config:
+            definition = {
+                "dependencies": processing_config.pop("packages"),
+            }
+        else:
+            raise ValueError(
+                "Either 'env_dict', 'env_yml_str' or 'packages' must be specified"
             )
```

Comment:
If the user specifies `env_dict` AND `packages` they may expect it to result in a combination of both? Should we show a warning that `env_dict` has precedence and ignoring `packages`?

```suggestion
        definition = None
        if "env_dict" in processing_config:
            definition = processing_config.pop("env_dict")

        if "env_yml_str" in processing_config:
            if definition:
                log.warn("Ignoring 'env_yml_str' configuration")
            else:
                import yaml

                definition = yaml.safe_load(processing_config.pop("env_yml_str"))

        if "packages" in processing_config:
            if definition:
                log.warn("Ignoring 'packages' configuration")
            else:
                definition = {
                    "dependencies": processing_config.pop("packages"),
                }

        if not definition:
            raise ValueError(
                "Either 'env_dict', 'env_yml_str' or 'packages' must be specified"
            )
```

Reply:
I'm ok with that too! Right now it ignores without letting them know

Reply:
> I'm ok with that too! Right now it ignores without letting them know

If I understand the thing here (e.g. specifying both `yml` + `dict`), we should be already handling it at the end (`**processed_config`).

Reply:
```
        environment = cls(
            environment_definition=definition,
            **processing_config,
        )
```

This will hanlde it because it errors out?

Should we check and be explicit about the real error?

Reply:
We use the same action mechanism for virtualenv too, but I understand it might not be super explicit. I think the next stage would be either isolate level or koldstart level schema validation like DBT does on the actual config object itself.

https://github.com/fal-ai/isolate/blob/4ec76dc69feff90f6951e88c6d6ac2634fc5d985/src/isolate/backends/virtualenv.py#L30-L38